### PR TITLE
cmake: rename libprinter-tool

### PR DESCRIPTION
**Pull request by junghans**
`libprinter-tool` seems a bit too generic as a library name plus no versioning.

### CODE REVIEW

**Comment by jjwilke** [comment]
Fine with me. @DavidPoliakoff is the author. I don't see any reason he would care?

**Comment by DavidPoliakoff** [comment]
Yeah I don't care. Just target against develop

**Comment by DavidPoliakoff** [comment]
Wait, this gets installed?

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -415,10 +415,12 @@ KOKKOS_ADD_ADVANCED_TEST( UnitTest_PushFinalizeHook_terminate
   if(KOKKOS_ENABLE_LIBDL)
 
     KOKKOS_ADD_LIBRARY(
-    printer-tool SHARED
+    kokkosprinter-tool SHARED
     SOURCES tools/printing-tool.cpp
     )
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt:420]: I think the issue is that we use `KOKKOS_ADD_LIBRARY`

Comment (DavidPoliakoff) [reply core/unit_test/CMakeLists.txt:420]: Is there an incantation of KOKKOS_ADD_LIBRARY that doesn't install?

Comment (DavidPoliakoff) [reply core/unit_test/CMakeLists.txt:420]: Or could we just remove https://github.com/kokkos/kokkos/commit/5621c99e1d011a2d5ec30ac1b8842e3eb4daa17a which changed this from add_library?

**Comment by dalg24** [comment]
Fixed in #3313.  Thanks for reporting this.

**Pull request closed without merging**